lib/internal/child_process.js

```diff
@@ -379,6 +379,9 @@ ChildProcess.prototype.spawn = function(options) {
 
       if (i > 0 && this.pid !== 0) {
         this._closesNeeded++;
+        if (stream._stdio) {
+          stream._stdio.emit('close');
+        }
```

Comment:
Shouldn’t this go in the `stream.socket.on('close', () => {` handler below?

Reply:
@addaleax I could do it. I fixed and push this code. Please check it.